### PR TITLE
HUD improvements

### PR DESCRIPTION
**Pull request by mbouron-gpsw**

### CODE REVIEW

File: libnodegl/node_hud.c [hunk 1]
```diff
@@ -1022,31 +1023,19 @@ static int widgets_init(struct ngl_node *node)
...
-        s->canvas.w = min_height * ar[0] / ar[1];
-        s->canvas.h = min_height;
-    }
+    const int latency_width  = get_widget_width(WIDGET_LATENCY);
+    const int memory_width   = get_widget_width(WIDGET_MEMORY);
+    const int activity_width = get_widget_width(WIDGET_ACTIVITY) * NB_ACTIVITY + WIDGET_MARGIN * (NB_ACTIVITY - 1);
+    const int drawcall_width = get_widget_width(WIDGET_DRAWCALL) * NB_DRAWCALL + WIDGET_MARGIN * (NB_DRAWCALL - 1);
+
+    s->canvas.w  = WIDGET_MARGIN * 2
+                 + NGLI_MAX(NGLI_MAX(NGLI_MAX(latency_width, memory_width), activity_width), drawcall_width);
```
Comment (cboesch-gpsw) [inline libnodegl/node_hud.c:1031]: extra vertical column space

File: libnodegl/node_hud.c [hunk 2]
```diff
@@ -1074,8 +1063,9 @@ static int widgets_init(struct ngl_node *node)
     }
 
     /* Draw-calls widgets in the bottom-right */
-    int x_drawcall =  -get_widget_width(WIDGET_DRAWCALL) * NB_DRAWCALL - WIDGET_MARGIN * NB_DRAWCALL;
-    const int y_drawcall = -get_widget_height(WIDGET_DRAWCALL) - WIDGET_MARGIN;
+    int x_drawcall = WIDGET_MARGIN;
+    //int x_drawcall =  -get_widget_width(WIDGET_DRAWCALL) * NB_DRAWCALL - WIDGET_MARGIN * NB_DRAWCALL;
```
Comment (cboesch-gpsw) [inline libnodegl/node_hud.c:1066]: left over?

File: libnodegl/node_hud.c [hunk 3]
```diff
@@ -1371,6 +1361,7 @@ static int hud_init(struct ngl_node *node)
     return 0;
 }
 
+
```
Comment (cboesch-gpsw) [inline libnodegl/node_hud.c:1363]: extra space

File: libnodegl/node_hud.c [hunk 4]
```diff
@@ -1172,7 +1172,7 @@ static void widgets_csv_report(struct ngl_node *node)
 
     ngli_bstr_clear(s->csv_line);
     /* Quoting to prevent locale issues with float printing */
-    ngli_bstr_printf(s->csv_line, "\"%f\"", s->last_refresh_time);
+    ngli_bstr_printf(s->csv_line, "\"%f\"", node->last_update_time);
```
Comment (cboesch-gpsw) [inline libnodegl/node_hud.c:1174]: This will cause a off-by-one, doesn't it?

File: libnodegl/node_hud.c [hunk 5]
```diff
@@ -1387,7 +1381,10 @@ static void hud_draw(struct ngl_node *node)
         return;
     }
 
-    if (s->need_refresh) {
+    double t = ngli_gettime_relative() / (double)1000000;
+    int need_refresh = fabs(t - s->last_refresh_time) >= s->refresh_rate_interval;
```
Comment (cboesch-gpsw) [inline libnodegl/node_hud.c:1385]: I think `t` and `need_refresh` can be `const`

File: libnodegl/node_hud.c [hunk 6]
```diff
@@ -1387,7 +1381,10 @@ static void hud_draw(struct ngl_node *node)
         return;
     }
 
-    if (s->need_refresh) {
+    double t = ngli_gettime_relative() / (double)1000000;
```
Comment (cboesch-gpsw) [inline libnodegl/node_hud.c:1384]: `1000000.` so you don't make a cast?

File: libnodegl/node_hud.c (hunk 2 above)
Comment (mbouron-gpsw) [reply libnodegl/node_hud.c:1066]: It is a left over. Updated.

File: libnodegl/node_hud.c (hunk 3 above)
Comment (mbouron-gpsw) [reply libnodegl/node_hud.c:1363]: Done.

File: libnodegl/node_hud.c (hunk 1 above)
Comment (mbouron-gpsw) [reply libnodegl/node_hud.c:1031]: Done.

File: libnodegl/node_hud.c (hunk 4 above)
Comment (mbouron-gpsw) [reply libnodegl/node_hud.c:1174]: It should not, this function is executed during the draw phase.

File: libnodegl/node_hud.c (hunk 6 above)
Comment (mbouron-gpsw) [reply libnodegl/node_hud.c:1384]: Ok.

**Review by cboesch-gpsw — APPROVED** [review]
LGTM, thanks.

**Pull request merged**